src/public/stageView.ts

```diff
@@ -0,0 +1,82 @@
+import { sendMessageToParent } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { SdkError } from '../public';
+import { FrameContexts } from './constants';
+
+/**
+ * Namespace to interact with the stage view specific part of the SDK.
+ */
+export namespace stageView {
+  /**
+   * Parameters to open a stage view.
+   */
+  export interface StageViewParams {
+    /**
+     * The application ID of the Teams application to be opened.
+     * Required
+     */
+    appId: string;
+
+    /**
+     * The URL of the content to display.
+     * Required
+     */
+    contentUrl: string;
+
+    /**
+     * The Teams app website URL.
+     * Not required but still used to populate an open website button in stage view.
+     */
+    websiteUrl?: string;
+
+    /**
+     * The name of the stage view.
+     * Delete seems unused
+     */
+    name?: string;
+
+    /**
+     * The entity ID.
+     * Not sure
+     */
+    entityId?: string;
+
+    /**
+     * The chat or channel ID.
+     * Not sure
+     */
+    threadId?: string;
+
+    /**
+     * The initatiator of the stage view request.
+     * Delete seems unused
+     */
+    source?: string;
+
+    /**
+     * The title to give the stage view.
+     * Used by the error dialog.  Should this be required?
+     */
+    title?: string;
+  }
+
+  /**
+   * @private
+   * Feature is under development
+   *
+   * Opens a stage view to display a Teams app
+   * @param stageViewParams The parameters to pass into the stage view.
+   * @param callback Callback that will be triggered once the stage view is closed.
+   *                 The callback takes as an argument an SdkError in case something happened (i.e.
+   *                 no permissions to execute the API)
+   */
+  export function openStageView(stageViewParams: StageViewParams, callback?: (sdkError?: SdkError) => void): void {
```

Comment:
@TrevorJoelHarris, @MarcinJuraszek, @erinha, so what do we wanna call this API? Should it just be `open`? Also, do we want to leave the namespace as is or change it to something simpler - e.g. `modal` or `stage`?

Reply:
Open is good, I agree with Marcin's comment above.

I suspect that eventually we will want to call this something other than stageView, but since we will still have the opportunity to change this up later as part of merging into 2.0 and I don't feel confident that modal is *better* than stageView, I am fine with leaving as is.